### PR TITLE
Update compute to latest version.

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Move to the latest compute package to be ready to support user-assigned
identities.  This required some other tweaks due to breaking changes.
Updated dep dependencies.

Thanks for your contribution! Please let us know what problems this PR addresses or what functionality it adds. Please link to other related issues and PRs to speed up review. Thank you!

### CODE REVIEW

File: compute/vm_disk.go [hunk 1]
```diff
@@ -9,7 +9,8 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2017-03-30/compute"
+	disks "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-04-01/compute"
```
Comment (joshgav) [inline compute/vm_disk.go]: Nice that multi-API allows us to do this :). It's not great that we have to totally rename but maybe this is the clearest way.

@marstr thoughts?

Comment (jhendrixMSFT) [reply compute/vm_disk.go]: The latest tag in the config file for compute doesn't actually include the latest compute APIs which support user-assigned identities, so we have to use disks from the previous release.

Comment (marstr) [reply compute/vm_disk.go]: This may be an example a scenario that would be improved if we published packages at a resource type level.

**Comment by flyinprogrammer** [comment]
https://github.com/Azure/azure-sdk-for-go/pull/2286

seems to be when `services/compute/mgmt/2018-06-01/compute/disks.go` was deleted.

there seems to be no replacement for the client in 2018-06-01, so i'm not really sure what the plan is to fix that?

**Comment by jhendrixMSFT** [comment]
There is no 2018-06-01 API version for disks, it comes from the 2018-04-01 ```compute``` package.

**Comment by jhendrixMSFT** [comment]
I just received a request from the compute folks to update our 2018-06-01 with their new package which will include disks.  This will be released as part of the v20 SDK.

**Pull request merged**